### PR TITLE
Team register backend

### PR DESCRIPTION
**Pull request by EetuHei**
- Team router, uses jwt for auth, has route to register a team. Returns the registered 

- Error handling with http errors, checks team name length 2-20 and checks if the team name already exists in the db.

- Team data has functions to get team and to register a team. Registering team uses transaction so if something fails it will not put anything to db.

- Registering team also add's player id and team id into team roster table when the registering is successful.

### CODE REVIEW

File: server/src/index.js [hunk 1]
```diff
@@ -16,6 +17,7 @@ app.use([
 ]);
 
 app.use('/api/v1/auth', authRouter);
+app.use('/api/v1/', teamRouter);
```
Comment (diegocasillas) [inline server/src/index.js]: Use here the router like `app.use('/api/v1/teams', teamRouter);` and change the register team route to simply `teamRouter.post('/', ...)`.

File: server/src/team/teamData.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    if (teamRegister && !!teamRegister.length) {
+      await trx('team_roster').insert({
+        playerId: captainId,
+        teamId: teamRegister[0]
+      });
+    }
+
+    const team = await trx('team')
+      .where('id', teamRegister[0])
```
Comment (diegocasillas) [inline server/src/team/teamData.js]: No biggie, but you can change this to `.where({ id: teamRegister[0] })` for consistency.

File: server/src/team/teamRouter.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        'Team name should be at least 2 characters and maximum of 20 characters'
+      )
+    );
+  }
+  if (isNaN(gameId)) {
+    return next(httpErrors(401, 'Game id can only be a number'));
+  }
+
+  try {
+    const teamRegister = await registerTeam(name, gameId, captainId);
```
Comment (diegocasillas) [inline server/src/team/teamRouter.js]: Rename this `teamRegister` to `team`.

File: server/src/team/teamRouter.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    if (!teamRegister) {
+      return next(httpErrors(500));
+    }
+    return res.status(201).json({ team: teamRegister });
```
Comment (diegocasillas) [inline server/src/team/teamRouter.js]: Return here `return res.data(201, { team });` instead (it formats the response wrapping it in a `data` object).

Comment (EetuHei) [reply server/src/team/teamRouter.js]: Fixed all these issues.

**Pull request merged**